src/directives/profile.js

```diff
@@ -39,11 +40,34 @@ ngeo.profileDirective = function() {
           goog.asserts.assert(goog.isDef(optionsAttr));
 
           var selection = d3.select(element[0]);
-          var profile, options, elevationData, poiData;
+          var profile, elevationData, poiData;
 
           scope.$watchCollection(optionsAttr, function(newVal) {
-            options = newVal;
+
+            var options = /** @type {ngeox.profile.ProfileOptions} */
+                (goog.object.clone(newVal));
+
             if (goog.isDef(options)) {
+
+              // proxy the hoverCallback in order to be able to call $apply
+              if (goog.isDef(options.hoverCallback)) {
+                var origHoverCallback = options.hoverCallback;
+                var hoverCallback = function(point) {
+                  origHoverCallback(point);
+                  scope.$applyAsync();
+                };
+                options.hoverCallback = hoverCallback;
```

Comment:
Also, reading the code, it is not clear why you need to use `$applySync` rather than `$apply` here. I do think this deserves a line of explanation.